src/main/java/io/debezium/connector/yugabytedb/transforms/StriimCompatible.java

```diff
@@ -0,0 +1,208 @@
+            if (!after.get(key).equals(before.get(key))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Pair<Schema, Struct> getUpdatedValueAndSchema(Schema schema, Struct value, List<String> primaryKeys) {
+        LOGGER.debug("Original Schema as json: " + io.debezium.data.SchemaUtil.asString(schema));
+        Schema updatedSchema = makeUpdatedSchema(schema);
+        LOGGER.debug("Updated schema as json: " + io.debezium.data.SchemaUtil.asString(updatedSchema));
+
+        List<String> allFields = getAllFieldsInOrder(schema.field("after").schema());
+
+        LOGGER.debug("Original value as json: {}", io.debezium.data.SchemaUtil.asDetailedString(value));
+        Struct newVal = new Struct(updatedSchema);
+        Struct metadata = makeMetadata(value, updatedSchema.field("metadata").schema());
+        newVal.put("columns", allFields);
+
+        switch ((String)value.get("op")) {
+            case "c": {
+               Map<String, Object> newValues = extractData((Struct) value.get("after"));
+
+                metadata.put("OperationName", "INSERT");
+                newVal.put("metadata", metadata);
+                newVal.put("data", convertToOrderedList(newValues, allFields));
+
+               LOGGER.debug("Update value as json: {}", io.debezium.data.SchemaUtil.asDetailedString(newVal));
+               return new org.yb.util.Pair<>(updatedSchema, newVal);
+            }
+            case "u": {
+                Map<String, Object> newValues = extractData((Struct) value.get("after"));
+                Map<String, Object> oldValues = extractData((Struct) value.get("before"));
+                removeNonPrimaryKeyValues(oldValues, primaryKeys);
+
+                metadata.put("OperationName", "UPDATE");
+                if (!comparePrimaryKeyValues(newValues, oldValues, primaryKeys)) {
+                    metadata.put("PK_UPDATE", true);
+                }
+                newVal.put("metadata", metadata);
+                newVal.put("data", convertToOrderedList(newValues, allFields));
+                newVal.put("before", convertToOrderedList(oldValues, allFields));
+
+                LOGGER.debug("Update value as json: {}", io.debezium.data.SchemaUtil.asDetailedString(newVal));
+                return new org.yb.util.Pair<>(updatedSchema, newVal);
+            }
+            case "d": {
+                Map<String, Object> oldValues = extractData((Struct) value.get("before"));
+                removeNonPrimaryKeyValues(oldValues, primaryKeys);
+
+                metadata.put("OperationName", "DELETE");
+                newVal.put("metadata", metadata);
+                newVal.put("data", convertToOrderedList(oldValues, allFields));
+
+                LOGGER.debug("Update value as json: {}", io.debezium.data.SchemaUtil.asDetailedString(newVal));
+                return new org.yb.util.Pair<>(updatedSchema, newVal);
+            }
+            default: {
```

Comment:
@suranjan Are we going to need a case for `READ (r)` as well?

Reply:
Yes, we should convert snapshot entry as well.

Reply:
Actually, there is no documentation on read records [here](https://www.striim.com/docs/en/postgresql-cdc.html#UUID-bb4c7685-acc1-6f6a-f4a2-afbfecf4febf). So, I am unsure about what to send in the `data`, `metadata` field.